compiler/rustc_middle/src/query/mod.rs

```diff
@@ -768,6 +767,28 @@ rustc_queries! {
         desc { |tcx| "comparing impl items against trait for `{}`", tcx.def_path_str(impl_id) }
     }
 
+    /// Given a trait `fn_def_id`, create and return the associated items list for that function.
+    query assoc_items_for_rpitits(fn_def_id: DefId) -> &'tcx [DefId] {
+        desc { |tcx| "creating associated items for rpits returned by `{}`", tcx.def_path_str(fn_def_id) }
+        cache_on_disk_if { fn_def_id.is_local() }
+        separate_provide_extern
+    }
+
+    /// Given an impl fn `impl_fn_def_id` create and return all the impl associated items corresponding to the rpits on the corresponding trait fn.
+    query impl_assoc_items_for_rpitits(impl_fn_def_id: DefId) -> &'tcx [DefId] {
+        desc { |tcx| "creating impl associated items for rpits returned by `{}`", tcx.def_path_str(impl_fn_def_id) }
+        cache_on_disk_if { impl_fn_def_id.is_local() }
+        separate_provide_extern
+    }
+
+    /// Given an opaque rpitit `opaque_ty_def_id`, creates and returns the corresponding associated item.
+    /// Returns `None` if it's an RPIT but not in trait.
+    query rpitit_associated_item(opaque_ty_def_id: LocalDefId) -> Option<LocalDefId> {
```

Comment:
The name "rpitit" is kinda vague here -- maybe let's call it `associated_item_for_impl_trait_in_trait`?

Reply:
Same for the rest of the queries -- we probably shouldn't use "rpitit" in names because of being an acronym. I think using impl_trait_in_trait may be longer but ultimately clearer.